xs2a-adapter-service-remote/src/main/java/de/adorsys/xs2a/adapter/service/impl/AccountInformationServiceImpl.java

```diff
@@ -0,0 +1,203 @@
+import de.adorsys.xs2a.adapter.service.account.TransactionsReport;
+import de.adorsys.xs2a.adapter.service.ais.*;
+import de.adorsys.xs2a.adapter.service.exception.NotAcceptableException;
+import de.adorsys.xs2a.adapter.service.model.*;
+import org.apache.commons.lang3.StringUtils;
+import org.mapstruct.factory.Mappers;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.ResponseEntity;
+
+import java.time.LocalDate;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class AccountInformationServiceImpl implements AccountInformationService {
+
+    private final AccountInformationClient client;
+    private final ConsentMapper consentMapper = Mappers.getMapper(ConsentMapper.class);
+    private final ConsentCreationResponseMapper creationResponseMapper = Mappers.getMapper(ConsentCreationResponseMapper.class);
+    private final ConsentInformationMapper consentInformationMapper = Mappers.getMapper(ConsentInformationMapper.class);
+    private final ConsentStatusResponseMapper statusResponseMapper = Mappers.getMapper(ConsentStatusResponseMapper.class);
+    private final ScaStatusResponseMapper scaStatusResponseMapper = Mappers.getMapper(ScaStatusResponseMapper.class);
+    private final BalanceReportMapper balanceReportMapper = Mappers.getMapper(BalanceReportMapper.class);
+    private final AccountListHolderMapper accountListHolderMapper = Mappers.getMapper(AccountListHolderMapper.class);
+    private final TransactionsReportMapper transactionsReportMapper = Mappers.getMapper(TransactionsReportMapper.class);
+    private final StartScaProcessResponseMapper scaProcessResponseMapper = Mappers.getMapper(StartScaProcessResponseMapper.class);
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    public AccountInformationServiceImpl(AccountInformationClient client) {
+        this.client = client;
+        objectMapper.setSerializationInclusion(JsonInclude.Include.NON_NULL);
+        objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+        objectMapper.registerModule(new JavaTimeModule());
+        objectMapper.disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS);
+
+    }
+
+    @Override
+    public GeneralResponse<ConsentCreationResponse> createConsent(RequestHeaders requestHeaders, Consents consents) {
+        ConsentsTO consentsTO = consentMapper.toConsentsTO(consents);
+        ResponseEntity<ConsentsResponse201TO> responseEntity = client.createConsent(requestHeaders.toMap(), consentsTO);
+        ConsentCreationResponse consentCreationResponse = creationResponseMapper.toConsentCreationResponse(responseEntity.getBody());
+        return new GeneralResponse<>(responseEntity.getStatusCodeValue(), consentCreationResponse, getHeaders(responseEntity.getHeaders()));
+    }
+
+    @Override
+    public GeneralResponse<ConsentInformation> getConsentInformation(String s, RequestHeaders requestHeaders) {
```

Comment:
Maybe we should rename `String s` to something more readable, what do you think?
The same story with the rest of the `String s` within this class.